### DISCUSSION TITLE
Change name of 'how-to' Module

### DISCUSSION THREAD

**Discussion by davehakkens** [ONEARMY/community-platform#3241] [Big Updates]
Wasn't planning to make a discussion topic for this. But why not, could use some community input :)
We developed a [how-to](https://community.preciousplastic.com/how-to) module, the goal of this module is to share **end result.** Orignally it was focussed on step-by-step guides (how-to's), througout the years this evolved and is now also used as a place to host and share files. For instance it also contains 3D files for moulds and machines. How-to's/guides are just a type of content. 
<img width="223" alt="afbeelding" src="https://github.com/ONEARMY/community-platform/assets/13672737/c033492f-05bb-480e-8704-5664ec4cd30d">

# Current Problem
- The name doesnt represent the full spectrum of the module
- The name is very nested in all the codebase, change might be complex
- Not sure what name is better suited.

# Whats needed next

- [ ] Understand the technical complexcity and implication of changing the name
- [ ] Find a better name. 

# Drop ideas for a name
So far we have (listed in preference):
1. Library
2. Database
3. Commons 
4. Collection
5. End-result

You have any preferences or suggestions?

**Comment by sigolenej** [comment]
Thanks for this proposal! 
I haven't been faced with user feedback on this as it was never fully my scope, but understand the context and the reason why a change is relevant. 

Out of your proposals, my favorite is library :) 
I don't find 2) database and 3) communs reflect what we have (or at least it's not clear for me). 

"Tutorials" could also be an option.

**Comment by m-azoubel** [comment]
I really like "Library" too, it's clearer and it reflects what we have. 
However, I'm a bit concerned it might be a very big change from what we have and that users might not take it simply.
Could we consider something along the lines of "How-to Library"?

- How-to Library
- How-to Guides
- Collaborative library
- Community library

**Comment by pizzaisdavid** [comment]
As I understand it, the prompt is "come up with a better word or phrase for _how-to_ in the URL https://community.preciousplastic.com/how-to (and also on the page title and navigation bar.)"

I am more techy person, so my first thought was https://community.preciousplastic.com/knowledge-base but I think that is not the most understandable for others.

instructables.com, a website for building hardware/coding, uses "projects" (https://www.instructables.com/projects) all of that content seems to be tutorials (first showing the end product then teaching how to make it.)

I looked briefly at the how-tos on precious plastic and didn't find any that are simply "here's a collection of files I made." I'm not sure if that is because there is a required minimum number of steps or if I didn't look hard enough. So if there is an example, I would be interested.

So the overall format would then be:
```
Projects
-> machines
-> moulds
-> products
-> ...guides?
```

Hey.. wait a second, it's kinda weird that "guides" is a subcategory of projects, especially since machines, moulds, and products are all guides on specific topics (at least to me and what I've seen.)

So I would rename "guides" to "other" (or rename it to "business" or similar but then items that don't match that would need to moved to another subcategory or new subcategory.)

So the proposal that does the smallest changes is:
```
Projects
-> machines
-> moulds
-> products
-> other
```

To bad other isn't plural though.

Closing argument: looking at https://community.preciousplastic.com/projects feels good.

But also, "projects" could be unclear that is it going to be essentially guides, so if you accept to change the subcategory "guides" to "other", then just make https://community.preciousplastic.com/tutorials or https://community.preciousplastic.com/guides

https://trends.google.com/trends/explore?date=now%201-d&geo=DE&q=guides,tutorials,guide,tutorial&hl=en

But since this word or phrase must exist next to "map" and "academy" in the navigation; I'd be confused "hey man, what's the difference between guides and academy?" well.. academy is official guides and guides is guides made by community members. "Ah okay" 

I think it would be natural to see projects next to map and academy. And projects are things... with reproducible steps.

```
academy -> learn
projects -> do
map -> meet
```

**Reply by davehakkens** [reply]
Thanks for the thoughts, very useful to get an outside view :)

One other variable to sketch a complete picture is that we also have a "research module" 
Currently used on [Project Kamp ](https://community.projectkamp.com/academy)but soon on Precious Plastic as well.
And you could consider them all "projects" as well.
Difference is
How-to =  End result shared 
Research = Ongoing process where the community can help out along the way

Knowing this you would still go for projects?

**Reply by pizzaisdavid** [reply]
I guess I am confused by this statement:

>  Orignally it was focussed on step-by-step guides (how-to's), througout the years this evolved and is now also used as a place to host and share files.

Could you give me an example of a how-to on precious plastic where they share some files but don't provide instructions?

So far, I've found them to always provide instructions, so to me, "How-to = End result shared" is really meaning "How-to = End result of reproducible steps shared." But I think I must be misunderstanding you.

But anyways, if the goal is to come up with a term that includes "Hey, I made this table. Here are the steps so you can make it too." and "Hey, I made this table, here are the model files. You got it from here, right?" and doesn't include ongoing research. Then maybe something like <ins>creations</ins>.

```
creations
-> machines
-> moulds
-> products
-> other
```

And then filtering by moulds, you would get both entries that are how-to and that are just files.

Creations implies finished to me so it doesn't overlap with ongoing research.

https://community.preciousplastic.com/creations

On this page, it is always someone sharing something they have made: a spreadsheet, a knife handle, a protocol for organizing an event.

If the goal is to come up with a better term for "Hey, I made this table. Here are the steps so you can make it too." I would agree with "tutorials" or "guides" but this must not be what you really want, right? Because in that case "how-to" still fits.

**Reply by davehakkens** [reply]
`Could you give me an example of a how-to on precious plastic where they share some files but don't provide instructions?
` 
Doest really exists since there is a min 3 step required to fit the "how-to" format. Changing the module-name would allow us to make a different template for someone that only really wants to share files (like [this](https://community.preciousplastic.com/how-to/nps_air-press-injector-v1) or [this](https://community.preciousplastic.com/how-to/flowerpot-mould) one)

But thanks for thinking of this, i'll give a think to "creations" to see if it fits our scenarios 
Just curious what would be against Library?

**Reply by pizzaisdavid** [reply]
> Just curious what would be against Library?

I would say, if the main focus of the community platform will be spent on the `/library` page, looking at how-tos and assets (for a lack of a better word) then it could be okay.

Steam, a platform for buying digital games uses the term library.

![Capture](https://github.com/ONEARMY/community-platform/assets/4391884/8d50410c-305b-4946-b0a3-95e8bc9cc73f)

Library communicates that it will be a collection of something. But it doesn't indicate what the items will be. So if I was on a community platform and saw Library in the navigation bar, I wouldn't know what it is without clicking on it.

In Steam's case, your library is empty until you buy your first game. And it is clear that library will refer to games, because Steam is a program for games.

To me, a community platform can have many different things (events, map with locations, research, guides, assets, academy, and q&a) related to a single topic. How to know the term library only refers to guides and assets? Maybe if that is the main content of the platform, then they would expect this.

Also, if you pick the term library, you'll probably still have to decide on a word for the singular noun. To explain concretely, on the `/how-to` page, there is a button with the text "Create a How-to" what will that say instead? It could say "Create" (intentionally never defining a singular version) or "Create an entry" (an entry to the library.)

**Comment by Christopher-UnmakePlastic** [comment]
Over at the Discord, Dave described the need for a new label this way-

"What do you all think about changing the name of how-to's into Library ... We've already been thinking about it for quite a while and should pave the way for the future of our **_documentation_**."

I like "Documentation"- the word used to describe the label, parsed of specificity, but still evoking the desired expectations.  Whatever the choice, if needing explanation, don't bury the explanation behind a click- use a sub-title, caption, explainer, whatever, _**at the first instance**_.    Imagine you're standing at a campus kiosk map, with all the buildings labeled with the names of people they honored- but not their function.   "Is the Hubble building Astronomy, or Applied Mathematics?"

**Reply by davehakkens** [reply]
Like documentation as well. However the term is a bit to broad.
Both the Academy and Research module could also be considered as documentation.

Do agree it would be important to give some explanation to these abstract terms anyway

**Comment by adriancoira** [comment]
Hei I also like library in the way that it is a "storage" space for different items. Seems to work now and in the future for the different iterations that could come. But it would need to have a nice categories that make it understandable for the user. I see a bit of conflict in the layers since research could also come as a category of the library
